recipes-app/iot2050-firmware-update/files/iot2050-firmware-update

```diff
@@ -272,6 +272,42 @@ class FirmwareUpdate(object):
         if firmware_md5_digest != firmware_flash_md5_digest:
             raise UpgradeError("Firmware digest verification failed")
 
+    def backup_rollback_firmware(self, firmware_len=0x8c0000,
+                                 path='/root/.rollback_fw/'):
```

Comment:
What if the current user is not root?

Better to backup the old firmware into the home of the current user.